frontend_v2/src/app/components/challenge/challengesettings/challengesettings.component.spec.ts

```diff
@@ -4,33 +4,714 @@ import { ChallengesettingsComponent } from './challengesettings.component';
+        observer.complete();
+        return { unsubscribe() {} };
+      })
+    );
+
+    component.editTermsAndConditions();
+    expect(globalService.showModal).toHaveBeenCalled();
+    component.apiCall(updatedTermsAndConditions);
+    expect(apiService.patchUrl).toHaveBeenCalled();
+    expect(component.challenge.terms_and_conditions).toEqual(updatedTermsAndConditions);
+    expect(component.updateView).toHaveBeenCalled();
+    expect(component.challenge.terms_and_conditions).toEqual(updatedTermsAndConditions);
+    expect(globalService.showToast).toHaveBeenCalledWith('success', expectedSuccessMsg, 5);
+  });
+
+  it('should handle the API error for `editTermsAndConditions` method', () => {
+    const updatedTermsAndConditions = 'Updated terms and conditions of challenge';
+    const expectedErrorMsg = {
+      error: 'Api error',
+    };
+    spyOn(apiService, 'patchUrl').and.returnValue(
+      new Observable((observer) => {
+        observer.error({ error: expectedErrorMsg.error });
+        observer.complete();
+        return { unsubscribe() {} };
+      })
+    );
+
+    component.editTermsAndConditions();
+    expect(globalService.showModal).toHaveBeenCalled();
+    component.apiCall(updatedTermsAndConditions);
+    expect(apiService.patchUrl).toHaveBeenCalled();
+    expect(globalService.showToast).toHaveBeenCalledWith('error', expectedErrorMsg);
+  });
+
+  it('should show modal and successfully edit evaluation script', () => {
+    const parameters = {
+      evaluation_script: 'evaluation_script',
+    };
+    const expectedSuccessMsg = 'The evaluation script is successfully updated!';
+    spyOn(apiService, 'patchFileUrl').and.returnValue(
+      new Observable((observer) => {
+        observer.next({ results: [{}] });
+        observer.complete();
+        return { unsubscribe() {} };
+      })
+    );
+
+    component.editEvaluationScript();
+    expect(globalService.showModal).toHaveBeenCalled();
+    component.apiCall(parameters);
+    expect(apiService.patchFileUrl).toHaveBeenCalled();
+    expect(globalService.showToast).toHaveBeenCalledWith('success', expectedSuccessMsg);
+  });
+
+  it('should handle the API error for `editEvaluationScript` method', () => {
+    const parameters = {
+      evaluation_script: 'evaluation_script',
+    };
+    const expectedErrorMsg = {
+      error: 'Api error',
+    };
+    spyOn(apiService, 'patchFileUrl').and.returnValue(
+      new Observable((observer) => {
+        observer.error({ error: expectedErrorMsg.error });
+        observer.complete();
+        return { unsubscribe() {} };
+      })
+    );
+
+    component.editEvaluationScript();
+    expect(globalService.showModal).toHaveBeenCalled();
+    component.apiCall(parameters);
+    expect(apiService.patchFileUrl).toHaveBeenCalled();
+    expect(globalService.showToast).toHaveBeenCalledWith('error', expectedErrorMsg);
+  });
+  
+  it('should show modal and successfully edit the challenge description', () => {
```

Comment:
change to `should show modal and successfully edit challenge description`

Reply:
Done!